File: backend/Models/Dairy.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace backend.Models
{
    public class Dairy
    {
        public Guid Id { get; set; }
        [Required]
        public Guid LivestockId { get; set; }
        [Required]
        public DateTime Date { get; set; }
        [Required]
        public decimal MilkYield { get; set; }
        public decimal FatContent { get; set; }
        public decimal ProteinContent { get; set; }
        public string? Quality { get; set; }
    }
}

File: frontend/src/types/dairy.ts
// dairy.ts
export interface Dairy {
    id: string;
    livestockId: string;
    date: string;
    session: string;
    milkYield: number;
    fatContent: number;
    snfContent: number;
    quality: string;
}

export interface DairyTrendPointDto {
    date: string;
    totalMilk: number;
    averageFat: number;
    averageProtein?: number;
    averageSnf: number;
}

export interface CreateDairyRequest {
    livestockId: string;
    date: string;
    session: string;
    milkYield: number;
    fatContent: number;
    snfContent: number;
    quality: string;
}

export interface FeedConsumption {
    id: string;
    livestockId?: string;
    date: string;
    feedType: string;
    quantity: number;
    cost: number;
    notes?: string;
}

export interface CreateFeedConsumptionRequest {
    livestockId?: string;
    date: string;
    feedType: string;
    quantity: number;
    cost: number;
    notes?: string;
}
